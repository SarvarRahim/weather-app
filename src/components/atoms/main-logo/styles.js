import styled from 'styled-components';

const StyledMainLogo = styled.div`
  & a {
    display: inline-block;
  }

  & > img {
    width: 100%;
  }

  @media only screen and (max-width: ${props => props.theme.devices.phone}) {
    width: 5.1rem;
  }
`;

export { StyledMainLogo };
