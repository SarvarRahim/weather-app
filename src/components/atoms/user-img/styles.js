import styled, { css } from 'styled-components';
import { getSizeBySizeType } from '@utils/style-helpers';

const StyledUserImg = styled.div`
  width: ${getSizeBySizeType({ large: 16, middle: 4, small: 2.4 })}rem;
  height: ${getSizeBySizeType({ large: 16, middle: 4, small: 2.4 })}rem;

  & > img {
    border-radius: 50%;
    width: 100%;
    height: 100%;
    object-fit: cover;
  }

  @media (max-width: 1024px) {
    ${props =>
      props.size === 'large' &&
      css`
        width: 12rem;
        height: 12rem;
      `}
  }
`;

export { StyledUserImg };
