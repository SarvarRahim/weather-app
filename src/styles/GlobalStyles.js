import { createGlobalStyle } from 'styled-components';

// IMPORT FONTS
import ManropeExtraLight from '../assets/fonts/Manrope-ExtraLight.ttf';
import ManropeLight from '../assets/fonts/Manrope-Light.ttf';
import Manrope from '../assets/fonts/Manrope-Regular.ttf';
import ManropeMedium from '../assets/fonts/Manrope-Medium.ttf';
import ManropeSemiBold from '../assets/fonts/Manrope-SemiBold.ttf';
import ManropeBold from '../assets/fonts/Manrope-Bold.ttf';
import ManropeExtraBold from '../assets/fonts/Manrope-ExtraBold.ttf';

export const GlobalStyles = createGlobalStyle`
@font-face {
    font-family: 'ManropeExtraLight';
    src: url(${ManropeExtraLight}) format('truetype');
    font-weight: 200;
  }
  
  @font-face {
    font-family: 'ManropeLight';
    src: url(${ManropeLight}) format('truetype');
    font-weight: 300;
  }
  
  @font-face {
    font-family: 'Manrope';
    src: url(${Manrope}) format('truetype');
    font-weight: 400;
  }
  
  @font-face {
    font-family: 'ManropeMedium';
    src: local('ManropeMedium'),
      url(${ManropeMedium}) format('truetype');
    font-weight: 500;
  }
  
  @font-face {
    font-family: 'ManropeSemiBold';
    src: url(${ManropeSemiBold}) format('truetype');
    font-weight: 600;
  }
  
  @font-face {
    font-family: 'ManropeBold';
    src: url(${ManropeBold}) format('truetype');
    font-weight: 700;
  }
  
  @font-face {
    font-family: 'ManropeExtraBold';
    src: url(${ManropeExtraBold}) format('truetype');
    font-weight: 800;
  }
    html, body, body div, span, object, iframe, h1, h2, h3, h4, h5, h6, p, blockquote, pre, abbr, address, cite, code, del, dfn, em, img, ins, kbd, q, samp, small, strong, sub, sup, var, b, i, dl, dt, dd, ol, ul, li, fieldset, form, label, legend, table, caption, tbody, tfoot, thead, tr, th, td, article, aside, figure, footer, header, menu, nav, section, time, mark, audio, video, details, summary {
        margin: 0;
        padding: 0;
        border: 0;
        font-size: 100%;
        font-weight: normal;
        vertical-align: baseline;
        background: transparent;
    }

    main, article, aside, figure, footer, header, nav, section, details, summary {display: block;}

    /* Handle box-sizing while better addressing child elements: */
    html {
        box-sizing: border-box;
        font-size: 62.5%; /* 1rem == 10px */
    }

    body, button {
        font-family: 'Manrope', sans-serif;
        font-size: 1.6rem;
    }

    *,
    *:before,
    *:after {
        box-sizing: inherit;
    }

    /* consider resetting the default cursor: https://gist.github.com/murtaugh/5247154 */

    /* Responsive images and other embedded objects */
    /* if you don't have full control over 'img' tags (if you have to overcome attributes), consider adding height: auto */
    img,
    object,

    /*
    Note: keeping IMG here will cause problems if you're using foreground images as sprites.
        In fact, it *will* cause problems with Google Maps' controls at small size.
        If this is the case for you, try uncommenting the following:

    #map img {
            max-width: none;
    }
    */

    /* force a vertical scrollbar to prevent a jumpy page */
    html

    /* we use a lot of ULs that aren't bulleted.
        you'll have to restore the bullets within content,
        which is fine because they're probably customized anyway */
    ul {list-style: none;}

    blockquote, q {quotes: none;}

    blockquote:before,
    blockquote:after,
    q:before,
    q:after {content: ''; content: none;}

    a {margin: 0; padding: 0; font-size: 100%; text-decoration: none; vertical-align: baseline; background: transparent;}

    del {text-decoration: line-through;}

    abbr[title], dfn[title] {border-bottom: 1px dotted #000; cursor: help;}

    /* tables still need cellspacing="0" in the markup */
    table {border-collapse: separate; border-spacing: 0;font: 100%;font-size: inherit;}
    th {font-weight: bold; vertical-align: bottom;}
    td {font-weight: normal; vertical-align: top;}

    hr {display: block; height: 1px; border: 0; border-top: 1px solid #ccc; margin: 1em 0; padding: 0;}

    input, select {vertical-align: middle;}

    pre {
        white-space: pre; /* CSS2 */
        white-space: pre-wrap; /* CSS 2.1 */
        white-space: pre-line; /* CSS 3 (and 2.1 as well, actually) */
        word-wrap: break-word; /* IE */
    }

    input[type="radio"] {
      vertical-align: text-bottom;
    }
    input[type="checkbox"] {
      vertical-align: bottom;
    }

    select, input, textarea {font: 99% sans-serif;}

    small {font-size: 85%;}

    strong {font-weight: bold;}

    td, td img {vertical-align: top;}

    /* Make sure sup and sub don't mess with your line-heights http://gist.github.com/413930 */
    sub, sup {font-size: 75%; line-height: 0; position: relative;}
    sup {top: -0.5em;}
    sub {bottom: -0.25em;}

    /* standardize any monospaced elements */

    /* hand cursor on clickable elements */
    .clickable,
    label,
    input[type=button],
    input[type=submit],
    input[type=file],
    button {cursor: pointer;}

    button[disabled] {
      user-select: none;
      pointer-events: none;
    }

    /* Webkit browsers add a 2px margin outside the chrome of form elements */
    button, input, select, textarea {margin: 0;}

    /* make buttons play nice in IE */
    button,
    input[type=button] {width: auto; overflow: visible;}

    /* let's clear some floats */
    .clearfix:after { content: " "; display: block; clear: both; }
`;
