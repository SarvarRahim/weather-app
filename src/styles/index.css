@font-face {
  font-family: 'ManropeExtraLight';
  src: local('ManropeExtraLight'),
    url('../assets/fonts/Manrope-ExtraLight.ttf') format('truetype');
  font-weight: 200;
}

@font-face {
  font-family: 'ManropeLight';
  src: local('ManropeLight'),
    url('../assets/fonts/Manrope-Light.ttf') format('truetype');
  font-weight: 300;
}

@font-face {
  font-family: 'Manrope';
  src: local('Manrope'),
    url('../assets/fonts/Manrope-Regular.ttf') format('truetype');
  font-weight: 400;
}

@font-face {
  font-family: 'ManropeMedium';
  src: local('ManropeMedium'),
    url('../assets/fonts/Manrope-Medium.ttf') format('truetype');
  font-weight: 500;
}

@font-face {
  font-family: 'ManropeSemiBold';
  src: local('ManropeSemiBold'),
    url('../assets/fonts/Manrope-SemiBold.ttf') format('truetype');
  font-weight: 600;
}

@font-face {
  font-family: 'ManropeBold';
  src: local('ManropeBold'),
    url('../assets/fonts/Manrope-Bold.ttf') format('truetype');
  font-weight: 700;
}

@font-face {
  font-family: 'ManropeExtraBold';
  src: local('ManropeExtraBold'),
    url('../assets/fonts/Manrope-ExtraBold.ttf') format('truetype');
  font-weight: 800;
}
