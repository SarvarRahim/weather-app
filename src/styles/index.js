export * from './GlobalStyles';
export * from './Theme';
export * from './flex-container';
