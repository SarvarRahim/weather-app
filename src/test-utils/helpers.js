import { render } from '@testing-library/react';
import React, { useRef } from 'react';
import { BrowserRouter, Router } from 'react-router-dom';
import { Provider as ReduxProvider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { theme } from '@styles/Theme';
import { store } from '@store';
import { Formik } from 'formik';
import { VideoContext } from '@courses/routes/courses-single/course.context';
import { createMemoryHistory } from 'history';

const AllTheProviders = ({ children, formikValues = {} }) => {
  const videoRef = useRef();
  return (
    <BrowserRouter>
      <ReduxProvider store={store}>
        <ThemeProvider theme={theme}>
          <Formik initialValues={formikValues}>
            <VideoContext.Provider value={videoRef}>
              {children}
            </VideoContext.Provider>
          </Formik>
        </ThemeProvider>
      </ReduxProvider>
    </BrowserRouter>
  );
};

export const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export const renderWithRouter = (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(<Router history={history}>{ui}</Router>),
    history
  };
};
