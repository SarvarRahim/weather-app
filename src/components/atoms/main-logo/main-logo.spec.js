import { customRender } from '@test-utils/helpers';
import { StyledMainLogo } from './styles';

describe('Atoms Context', () => {
  test('Should render MainLogo component', async () => {
    const { asFragment } = customRender(<StyledMainLogo />);
    expect(asFragment()).toMatchSnapshot();
  });
});
