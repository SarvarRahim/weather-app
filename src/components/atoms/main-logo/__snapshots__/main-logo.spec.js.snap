// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms Context Should render MainLogo component 1`] = `
<DocumentFragment>
  .c0 a {
  display: inline-block;
}

.c0 > img {
  width: 100%;
}

@media only screen and (max-width:1024px) {
  .c0 {
    width: 5.1rem;
  }
}

<div
    class="c0"
  />
</DocumentFragment>
`;
